Make test targets invisible to GN to enable building with system abseil

--- a/third_party/abseil-cpp/BUILD.gn	2022-06-14 12:31:29.330129700 +0200
+++ b/third_party/abseil-cpp/BUILD.gn	2022-06-23 22:05:35.293708600 +0200
@@ -106,15 +106,6 @@
   visibility = [ ":absl" ]
 }
 
-group("default") {
-  deps = [
-    "absl/types:any",
-    "absl/types:bad_any_cast",
-    "absl/types:bad_optional_access",
-    "absl/types:optional",
-    "absl/types:span",
-  ]
-}
 
 config("absl_include_config") {
   include_dirs = [ "." ]
@@ -174,63 +165,3 @@
   }
 }
 
-if (build_with_chromium) {
-  import("//testing/test.gni")
-
-  if (!is_component_build) {
-    test("absl_tests") {
-      testonly = true
-      deps = [
-        "absl/algorithm:algorithm_test",
-        "absl/algorithm:container_test",
-        "absl/base:config_test",
-        "absl/base:prefetch_test",
-        "absl/cleanup:cleanup_test",
-        "absl/container:inlined_vector_test",
-        "absl/container:node_slot_policy_test",
-        "absl/container:sample_element_size_test",
-        "absl/hash:hash_test",
-        "absl/hash:low_level_hash_test",
-        "absl/memory:memory_test",
-        "absl/meta:type_traits_test",
-        "absl/profiling:exponential_biased_test",
-        "absl/profiling:periodic_sampler_test",
-        "absl/status:statusor_test",
-        "absl/strings:ascii_test",
-        "absl/strings:cord_data_edge_test",
-        "absl/strings:cord_rep_btree_navigator_test",
-        "absl/strings:cord_rep_btree_reader_test",
-        "absl/strings:cord_rep_btree_test",
-        "absl/strings:cord_rep_crc_test",
-        "absl/strings:cordz_functions_test",
-        "absl/strings:cordz_info_statistics_test",
-        "absl/strings:cordz_info_test",
-        "absl/strings:cordz_test",
-        "absl/strings:cordz_update_scope_test",
-        "absl/strings:cordz_update_tracker_test",
-        "absl/strings:match_test",
-        "absl/strings:str_replace_test",
-        "absl/strings:string_view_test",
-        "absl/types:optional_test",
-        "absl/types:variant_test",
-        "//third_party/googletest:gtest_main",
-      ]
-    }
-  }
-
-  test("absl_hardening_tests") {
-    testonly = true
-    sources = [ "absl_hardening_test.cc" ]
-    deps = [
-      "//base/test:run_all_unittests",
-      "//third_party/abseil-cpp/absl/base:config",
-      "//third_party/abseil-cpp/absl/container:fixed_array",
-      "//third_party/abseil-cpp/absl/container:inlined_vector",
-      "//third_party/abseil-cpp/absl/strings",
-      "//third_party/abseil-cpp/absl/types:optional",
-      "//third_party/abseil-cpp/absl/types:span",
-      "//third_party/abseil-cpp/absl/types:variant",
-      "//third_party/googletest:gtest",
-    ]
-  }
-}
--- a/BUILD.gn	2022-06-14 12:31:29.330129700 +0200
+++ b/BUILD.gn	2022-06-23 22:05:35.293708600 +0200
@@ -106,9 +106,6 @@
     "//url:url_unittests",
   ]
 
-  if (!is_component_build) {
-    deps += [ "//third_party/abseil-cpp:absl_tests" ]
-  }
 
   if (!is_android && !is_chromecast) {
     deps += [
